pkg/asset/internal/templates.go

```diff
@@ -729,15 +771,65 @@ spec:
       - name: ssl-certs-host
         hostPath:
           path: /usr/share/ca-certificates
-      - name: etc-kubernetes
-        hostPath:
-          path: /etc/kubernetes
+        name: ssl-certs-host
+      - name: kubeconfig
+        secret:
+          secretName: service-account-kubeconfig
   updateStrategy:
     rollingUpdate:
       maxUnavailable: 1
     type: RollingUpdate
 `)
 
+var ProxyServiceAccount = []byte(`apiVersion: v1
+kind: ServiceAccount
+metadata:
+  namespace: kube-system
+  name: kube-proxy
+`)
+
+var ProxyClusterRoleBinding = []byte(`apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: kube-proxy
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:node-proxier # Automatically created system role.
+subjects:
+- kind: ServiceAccount
+  name: kube-proxy
+  namespace: kube-system
+`)
+
+// ServiceAccountKubeConfig is used for pods that don't have access to the
```

Comment:
Can you clarify "used for"? This still feels odd and kinda opaque. So before we relied on the host's kubeconfig which contained both endpoint info and credentials. With the service account approach you use an alternate kubeconfig which specifies the endpoint info and the local service account in the pod.

What's the tie in to these environment variables?

```
KUBERNETES_SERVICE_PORT=443
KUBERNETES_PORT=tcp://10.3.0.1:443
```

Reply:
Clients compiled with client-go require the `KUBERNETES_SERVICE_PORT` and `KUBERNETES_PORT` to use the in-cluster config.

https://github.com/kubernetes/client-go/blob/kubernetes-1.8.2/rest/config.go#L318-L321

These aren't exposed to kube-proxy (since kube proxy sets up the endpoint) or the checkpointer, since it runs as a static pod.

This work:

```
// ServiceAccountKubeConfig instructs clients to use their service account token, but unlike an in-cluster client
// doesn't rely on the `KUBERNETES_SERVICE_PORT` and `KUBERNETES_PORT` to determine the API
// servers address. This kubeconfig is used bootstrapping pods that might not have access to these env vars,
// such as kube-proxy, which sets up the endpoint (chicken and egg), and the checkpointer, which needs to run
// as a static pod even if the API server isn't available.
```